plugins/parodos/src/components/workflow/WorkflowsTable.tsx

```diff
@@ -58,13 +93,46 @@ const formatDate = new Intl.DateTimeFormat('en', {
 
 export const WorkflowsTable: React.FC<{
   projectId: string;
-  workflows: WorkflowStatus[];
+  workflows: ProjectWorkflow[];
 }> = ({ projectId, workflows }) => {
+  const filterIconRef = useRef<HTMLButtonElement>(null);
   const classes = useStyles();
   const [search, setSearch] = useState('');
+  const [statusFilter, setStatusFilter] =
+    useState<ProjectWorkflow['workStatus']>();
+  const [openFilter, setOpenFilter] = useState(false);
   const getWorkDefinitionBy = useStore(state => state.getWorkDefinitionBy);
 
+  const handleFilterToggle = () => setOpenFilter(prevOpen => !prevOpen);
+  const handleFilterClose = (event: React.MouseEvent<Node, MouseEvent>) => {
+    if (
+      filterIconRef.current &&
+      filterIconRef.current.contains(event.target as Node)
+    ) {
+      return;
+    }
+
+    setOpenFilter(false);
+  };
+  const handleChangeFilter = (
+    event: React.MouseEvent<Node, MouseEvent>,
+    filter?: ProjectWorkflow['workStatus'],
+  ) => {
+    handleFilterClose(event);
```

Comment:
can we do this at once?

Reply:
What do you mean?

Reply:
I thought it was 2 setState calls.   It's only 1.  My bad.